query-engine/core/src/interpreter/query_interpreters/read.rs

```diff
@@ -197,14 +206,25 @@ fn read_related<'a, 'b>(
                 let parent_fields = &parent_result.field_names;
                 let mut additional_records = vec![];
 
+                let mut the_map: HashMap<Vec<&PrismaValue>, Vec<&Record>> = HashMap::new();
```

Comment:
I would prefer a better naming like `id_cache` or whatever you deem more fitting.